Transports/com.community.netcode.transport.facepunch/Runtime/FacepunchTransport.cs

```diff
@@ -203,11 +218,16 @@ void IConnectionManager.OnDisconnected(ConnectionInfo info)
                 Debug.Log($"[{nameof(FacepunchTransport)}] - Disconnected Steam user {info.Identity.SteamId}.");
         }
 
-        void IConnectionManager.OnMessage(IntPtr data, int size, long messageNum, long recvTime, int channel)
+        unsafe void IConnectionManager.OnMessage(IntPtr data, int size, long messageNum, long recvTime, int channel)
         {
-            byte[] payload = new byte[size];
-            Marshal.Copy(data, payload, 0, size);
-            InvokeOnTransportEvent(NetworkEvent.Data, ServerClientId, new ArraySegment<byte>(payload, 0, size), Time.realtimeSinceStartup);
+            EnsurePayloadCapacity(size);
+
+            fixed (byte* payload = payloadCache)
+            {
+                UnsafeUtility.MemCpy(payload, (byte*)data, size);
```

Comment:
This could be just kept as Marshal.Copy and unsafe code is not giving us any advantage here right?

Reply:
@LukeStampfli I would say if there is no advantage to doing so, the NetCode package itself would be using Marshal.Copy instead of native Unity MemCpy.